ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -358,27 +360,323 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
         }
     }
 
+    void assembleForStaggeredScheme(
+        double const t,
+        std::vector<double>& local_M_data,
+        std::vector<double>& local_K_data,
+        std::vector<double>& local_b_data,
+        LocalCoupledSolutions const& coupled_xs) override
+    {
+        if (coupled_xs.process_id == hydraulic_process_id)
+        {
+            assembleHydraulicEquation(t, local_M_data, local_K_data,
+                                      local_b_data, coupled_xs);
+        }
+        else
+        {
+            // Component-wise
+            assembleComponentTransportEquation(t, local_M_data, local_K_data,
+                                               local_b_data, coupled_xs);
+        }
+    }
+
+    void assembleHydraulicEquation(double const t,
+                                   std::vector<double>& local_M_data,
+                                   std::vector<double>& local_K_data,
+                                   std::vector<double>& local_b_data,
+                                   LocalCoupledSolutions const& coupled_xs)
+    {
+        auto local_p = Eigen::Map<const NodalVectorType>(
+            coupled_xs.local_coupled_xs[hydraulic_process_id].data(),
+            pressure_size);
+        int const transport_process_id = 1;
+        auto local_C = Eigen::Map<const NodalVectorType>(
+            coupled_xs.local_coupled_xs[transport_process_id].data(),
+            concentration_size);
+        auto local_C0 = Eigen::Map<const NodalVectorType>(
+            coupled_xs.local_coupled_xs0[transport_process_id].data(),
+            concentration_size);
+
+        auto const dt = coupled_xs.dt;
+
+        auto local_M = MathLib::createZeroedMatrix<LocalBlockMatrixType>(
+            local_M_data, pressure_size, pressure_size);
+        auto local_K = MathLib::createZeroedMatrix<LocalBlockMatrixType>(
+            local_K_data, pressure_size, pressure_size);
+        auto local_b = MathLib::createZeroedVector<LocalSegmentVectorType>(
+            local_b_data, pressure_size);
+
+        unsigned const n_integration_points =
+            _integration_method.getNumberOfPoints();
+
+        SpatialPosition pos;
+        pos.setElementID(_element.getID());
+
+        auto const& b = _process_data.specific_body_force;
+
+        MaterialLib::Fluid::FluidProperty::ArrayType vars;
+
+        for (std::size_t ip(0); ip < n_integration_points; ++ip)
```

Comment:
should be `unsigned`, same as `n_integration_points`, no need to increase the integer's size.

Reply:
Fixed.